### DISCUSSION TITLE
Livewire.getByName | how do I get the wire obj by comp name?

### DISCUSSION THREAD

**Discussion by levaweb** [livewire/livewire#7367] [1. Help] [answered]
In JS I can get the wire object of the first() component on the page by using the global Livewire object , but how do I get the wire object of my component by-name? (I don't know its ID)

For example: my livewire component class is SideMenu, other than .first() none of the methods below works: I could not find much online about this. Thanks
https://livewire.laravel.com/docs/javascript#the-wire-object

```
let lw_sidebar = Livewire.first();  // OK
lw_sidebar.$refresh();   // OK

let lw_sidebar = Livewire.getByName("SideMenu");  // ERR
let lw_sidebar = Livewire.getByName("side-menu");  // ERR
var t1=JSON.stringify(lw_sidebar);  // ERR - trying to look at the working object lw_sidebar 



```

**Comment by WilliamPinto-Olmos** [comment]
Hey @alex32, i got two questions, are you getting an actual error or the function just returns an empty array? and  which version of livewire are you using? (Livewire.getByName was introduced in V3)

Also keep in mind that if your component is in a nested folder you need to call that folder/namespace too. For example, given the next component inside `SomeNamespace` folder.

```php
namespace App\Livewire\SomeNamespace;

use Livewire\Component;

class SomeComponent extends Component
{
    public function render()
    {
        return view('livewire.some-namespace.some-component');
    }
}
```
you need to call it like `Livewire.getByName('some-namespace.some-component')`: 

```html
<div x-data x-init="
    console.log(Livewire.getByName('some-namespace.some-component')[0]);
">
    ...
</div>
```

If you inspect `console.log(Livewire.getByName('some-namespace.some-component')[0].__instance);` you'll see something like this:

![Screenshot from 2023-11-23 09-54-16](https://github.com/livewire/livewire/assets/60890527/bc6ec7cb-8459-4ef8-89e6-62174b4611e0)

As you can see the property `name` include the `some-namespace`.

**Reply by levaweb** [reply]
Hi @WilliamPinto-Olmos thanks for your help.

Q1: No, my class is not nested into another folder (see scr-shot)
Q2: I use Laravel 10 + Lwire 3. If I try to get **wire** object using _getByName_() I get the attached error (method not found - as _lw_sidebar_ seems null.)

Not sure why Lwire doesn't like a simple call by name.. Thanks

```
    success: function(ret) {    
                        if (ret.type=="ERR") { toast (ret.msg); }    
                            else {  
                                //let lw_sidebar = Livewire.first();  // OK

                        console.log(Livewire.getByName('side-menu')[0].__instance); // Test Log

                                lw_sidebar= Livewire.getByName("side-menu"); // return empty array?
                                lw_sidebar.$refresh();  // ERR     

```

```
namespace App\Livewire;

use Livewire\Component;
use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Log;

class SideMenu extends Component
{
    public $firstProj;  // passed in blade to livewire

    public function render() {

        $uid=Auth::id();
```
https://snipboard.io/zyBuPg.jpg
https://snipboard.io/Uo0OsT.jpg

**Reply by WilliamPinto-Olmos** [reply, marked as answer]
hey @alex32 try `Livewire.getByName("side-menu")[0].$refresh()` instead of `Livewire.getByName("side-menu").$refresh()` since the `getByName()` returns a Proxy object with all the components with the name `side-menu` (in this case), so you need to access the first of them.

```js
// wont work
lw_sidebar= Livewire.getByName("side-menu"); 
lw_sidebar.$refresh();  
```

```js
// Should work
lw_sidebar = Livewire.getByName("side-menu")[0]; 
lw_sidebar.$refresh();  
```

Just tested it and it should work, but let me know if you still get the error

**Reply by levaweb** [reply]
Awesome!
It worked. Many thanks

**Reply by WilliamPinto-Olmos** [reply]
Glat that it helped:)
